python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -725,6 +740,30 @@ def do_evaluate(
                     else prefix.obj,
                 )
             )
+        elif self.name == pl_expr.StringFunction.Replace:
+            column, target, repl = columns
+            n, literal = self.options
+            if target.is_scalar and repl.is_scalar:
+                return Column(
+                    plc.strings.replace.replace(
+                        column.obj, target.obj_scalar, repl.obj_scalar, maxrepl=n
+                    )
+                )
+            raise NotImplementedError("replace")
+        elif self.name == pl_expr.StringFunction.ReplaceMany:
+            column, target, repl = columns
+            if repl.is_scalar:
+                # Try broadcasting repl scalar to column
+                # to support many-to-one replace
+                repl = Column(
+                    plc.Column.from_scalar(repl.obj_scalar, target.obj.size()),
+                    is_sorted=plc.types.Sorted.YES,
+                    order=plc.types.Order.ASCENDING,
+                    null_order=plc.types.NullOrder.BEFORE,
+                )
```

Comment:
libcudf supports the case of `repl.size() == 1` and `target.size() != 1` transparently, so:
```suggestion
```